File: threads/src/main/java/im/threads/internal/model/ConsultInfo.java
package im.threads.internal.model;

import androidx.annotation.NonNull;

import com.google.gson.JsonObject;

public final class ConsultInfo {

    private final String name;
    private final String id;
    private final String status;
    private final String organizationUnit;
    private final String photoUrl;
    private final String role;

    @Deprecated
    public ConsultInfo(String name,
                       String id,
                       String status,
                       String organizationUnit,
                       String photoUrl) {
        this.name = name;
        this.id = id;
        this.status = status;
        this.organizationUnit = organizationUnit;
        this.photoUrl = photoUrl;
        this.role = null;
    }

    public ConsultInfo(String name,
                       String id,
                       String status,
                       String organizationUnit,
                       String photoUrl,
                       String role) {
        this.name = name;
        this.id = id;
        this.status = status;
        this.organizationUnit = organizationUnit;
        this.photoUrl = photoUrl;
        this.role = role;
    }

    @NonNull
    @Override
    public String toString() {
        return "ConsultInfo{" +
                "name='" + name + '\'' +
                ", id=" + id +
                ", status='" + status + '\'' +
                ", organizationUnit='" + organizationUnit + '\'' +
                ", photoUrl='" + photoUrl + '\'' +
                ", role='" + role + '\'' +
                '}';
    }

    public String getName() {
        return name;
    }

    public String getId() {
        return id;
    }

    public String getStatus() {
        return status;
    }

    public String getOrganizationUnit() {
        return organizationUnit;
    }

    public String getPhotoUrl() {
        return photoUrl;
    }

    public String getRole() {
        return role;
    }

    public JsonObject toJson() {
        JsonObject jsonObject = new JsonObject();
        jsonObject.addProperty("name", name);
        jsonObject.addProperty("status", status);
        jsonObject.addProperty("id", id);
        jsonObject.addProperty("photoUrl", photoUrl);
        jsonObject.addProperty("role", role);
        return jsonObject;
    }
}

File: threads/src/main/java/im/threads/business/models/ConsultInfo.kt
package im.threads.business.models

import android.os.Parcelable
import com.google.gson.JsonObject
import kotlinx.parcelize.Parcelize

@Parcelize
data class ConsultInfo(
    val name: String? = null,
    val id: String? = null,
    val status: String? = null,
    val organizationUnit: String? = null,
    val photoUrl: String? = null,
    val role: String? = null
) : Parcelable {

    override fun toString(): String {
        return """ConsultInfo{
            name='$name',
            id=$id,
            status='$status',
            organizationUnit='$organizationUnit',
            photoUrl='$photoUrl',
            role='$role'
            }
        """.trim()
    }

    fun toJson(): JsonObject {
        return JsonObject().apply {
            addProperty("name", name)
            addProperty("status", status)
            addProperty("id", id)
            addProperty("photoUrl", photoUrl)
            addProperty("role", role)
        }
    }
}
